plugin/db/mysql/restore.go

```diff
@@ -319,10 +320,10 @@ func (driver *Driver) GetLatestBackupBeforeOrEqualTs(ctx context.Context, backup
 		validBackupList = append(validBackupList, b)
 	}
 
-	return getLatestBackupBeforeOrEqualBinlogCoord(validBackupList, *targetBinlogCoordinate, mode)
+	return getLatestBackupBeforeOrEqualBinlogCoord(validBackupList, *targetBinlogCoordinate, mode, driver.mysqlutil, driver.binlogDir)
```

Comment:
Let's change it to a method of `Driver`: `func (driver *Driver) getLatestBackupBeforeOrEqualBinlogCoord`